src/lazy/text/buffer.rs

```diff
@@ -533,7 +556,9 @@ impl<'data> TextBufferView<'data> {
     }
 
     /// Matches an optional annotation sequence and a trailing value.
-    pub fn match_annotated_value(self) -> IonParseResult<'data, MatchedRawTextValue<'data>> {
+    pub fn match_annotated_value(
+        self,
+    ) -> IonParseResult<'top, MatchedRawTextValue<'top, TextEncoding_1_0>> {
```

Comment:
🗺️ `MatchedRawTextValue` is now generic over an `E: TextEncoding<'top>`, which allows it to contain a cache of raw value expressions of that encoding.